addons/mail/static/src/models/composer/composer.js

```diff
@@ -852,6 +852,11 @@ function factory(dependencies) {
         }),
         attachments: many2many('mail.attachment', {
             inverse: 'composers',
+            // FIXME: this field should be readonly.
+            // The _reset function update this field with a [[unlink-all]] to
+            // ensure attachment are removed.
+            // Since readonly field can't be updated at all, this unlink tigger
+            // an error
```

Comment:
Ok now I understand, well then the field should not be read-only and you can remove this whole comment.

Also functionally it doesn't make sense for attachments to be read-only on composer, of course the user should be able to add/remove attachments wile composing a message.